src/WinGetUtil/Exports.cpp

```diff
@@ -21,8 +22,13 @@ extern "C"
     {
         THROW_HR_IF(E_INVALIDARG, !logPath);
 
-        static std::once_flag s_initLogging;
-        std::call_once(s_initLogging, []() {
+        thread_local AppInstaller::ThreadLocalStorage::ThreadGlobals threadGlobals;
+        thread_local std::once_flag initLogging;
+
+        std::call_once(initLogging, []() {
+            std::unique_ptr<AppInstaller::ThreadLocalStorage::PreviousThreadGlobals> previous = threadGlobals.SetForCurrentThread();

```

Comment:
While this may solve the problem when the code stays on one thread, it may make other problems that we are not aware of.  I would rather we fix it completely by creating a set of APIs that give more explicit control over the thread globals so that callers can actually control what is happening.